src/libraries/System.Security.Cryptography.ProtectedData/tests/SpanProtectedDataTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Security.Cryptography.ProtectedDataTests;
+
+public class SpanProtectedDataTests : ProtectedDataTests
+{
+    protected override byte[] Protect(byte[] plain, byte[]? entropy, DataProtectionScope scope)
+    {
+        ReadOnlySpan<byte> inputSpan = plain;
+        ReadOnlySpan<byte> entropySpan = entropy;
+        Span<byte> destination = stackalloc byte[400];
+        int written = ProtectedData.Protect(inputSpan, scope, destination, entropySpan);
+        return destination.Slice(0, written).ToArray();
+    }
+
+    protected override byte[] Unprotect(byte[] encrypted, byte[]? entropy, DataProtectionScope scope)
+    {
+        ReadOnlySpan<byte> inputSpan = encrypted;
+        ReadOnlySpan<byte> entropySpan = entropy;
+        Span<byte> destination = stackalloc byte[encrypted.Length];
+        int written = ProtectedData.Unprotect(inputSpan, scope, destination, entropySpan);
+        return destination.Slice(0, written).ToArray();
+    }
+
+    [Fact]
+    public void ZeroBufferThrows()
+    {
+        Assert.Throws<ArgumentException>(() =>
+            ProtectedData.Protect([1, 2, 3], DataProtectionScope.CurrentUser, Span<byte>.Empty));
+    }
+
+    [Theory]
+    [InlineData(-1, false)]
+    [InlineData(0, true)]
+    [InlineData(1, true)]
+    public void NearCorrectSizeBufferTests(int delta, bool success)
+    {
+        if (success)
```

Comment:
To make it more clear where the exception is coming from, consider

```C#
Span<byte> buffer = stackalloc byte[400];
ReadOnlySpan<byte> input = [1, 2, 3];
DataProtectionScope scope = DataProtectionScope.CurrentUser;

int written = ProtectedData.Protect(input, scope, buffer);
Span<byte> sliced = buffer.Slice(0, written + delta);

if (delta > 0)
{
    int written2 = ProtectedData.Protect(input, scope, sliced);
    Assert.Equal(written, written2);
}
else
{
    try
    {
        ProtectedData.Protect(input, scope, sliced);
        Assert.Fail("An exception should have been thrown because destination is too small");
    }
    catch (ArgumentException e)
    {
        Assert.Equal("destination", e.ParamName);
        Assert.IsType<ArgumentException>(e);
    }
}
```

What you have certainly works, but it takes a bit of reading to understand where the method is testing.

The try+Fail/catch can be avoided by changing from a stackalloc to a byte[] so it can be captured; but I went ahead and wrote it out in the "stackalloc is fun" way.